src/main/java/com/zufar/onlinestore/payment/controller/PaymentController.java

```diff
@@ -25,9 +31,13 @@ public class PaymentController implements PaymentApi {
     public static final String PAYMENT_URL = "/api/v1/payment";
 
     private final PaymentService paymentService;
+    private final PaymentEventService paymentEventService;
 
     @PostMapping
-    public ResponseEntity<PaymentDetailsWithTokenDto> paymentProcess(@RequestBody @Valid CreatePaymentDto paymentRequest) throws StripeException {
+    public ResponseEntity<PaymentDetailsWithTokenDto> paymentProcess(@RequestBody CreatePaymentDto paymentRequest) throws StripeException {
+        if (paymentRequest == null) {
```

Comment:
> It would be better, hah?
> 
> ```
> @RequestBody @NotNull @Valid CreatePaymentDto paymentRequest
> ```

это не работает в случае если мы имплементимся от интерфейса то нам нужно логику ограничений выносить на уровень интерфейса (родителя) иначе мы нарушаем Liskov принцип из SOLID, и это бы еще ничего, но мы постоянно будем падать с ошибкой при попытке дергнуть эндпоинт. Не вижу другого выхода пофиксить это, поскольку мы решили не выносить какую либо валидацию на уровень интерфейса

Reply:
make comments in English please